### PR TITLE
Make reporter buffer size and period configurable.

### PR DESCRIPTION
**Pull request by KevinMGranger**
This is one part of the work from #9.

I'm taking over working on these changes from Mark.

### CODE REVIEW

File: zipkin/include/zipkin/tracer.h [hunk 1]
```diff
@@ -50,8 +50,10 @@ typedef std::unique_ptr<Reporter> ReporterPtr;
  * service.
  * @return a Reporter object.
  */
-ReporterPtr makeHttpReporter(const char *collector_host,
-                             uint32_t collector_port);
+ReporterPtr makeHttpReporter(
+    const char *collector_host, uint32_t collector_port,
+    SteadyClock::duration reporting_period = std::chrono::milliseconds{500},
```
Comment (KevinMGranger) [inline zipkin/include/zipkin/tracer.h]: Are we okay with repeating the defaults here as well as in the Options struct, or do we want to put these somewhere shared?

File: zipkin/src/span_buffer.h [hunk 2]
```diff
@@ -40,6 +40,12 @@ class SpanBuffer {
    */
   bool addSpan(const Span &span);
 
+  /**
+   * @return returns the number of spans that can be held in currently allocated
+   * storage.
+   */
+  uint64_t spanCapacity() { return span_buffer_.capacity(); }
```
Comment (rnburn) [inline zipkin/src/span_buffer.h]: Can this be const?

File: zipkin/src/zipkin_reporter_impl.h [hunk 3]
```diff
@@ -28,7 +28,9 @@ class ReporterImpl : public Reporter {
    *
    * @param transporter The Transporter to be associated with the reporter.
    */
-  explicit ReporterImpl(TransporterPtr &&transporter);
+  explicit ReporterImpl(TransporterPtr &&transporter,
+                        SteadyClock::duration reporting_period,
+                        size_t max_buffered_spans);
```
Comment (rnburn) [inline zipkin/src/zipkin_reporter_impl.h]: We can remove explicit now.

File: zipkin/include/zipkin/tracer.h (hunk 1 above)
Comment (rnburn) [reply zipkin/include/zipkin/tracer.h]: 👍 Like the idea of putting defaults somewhere shared.

**Comment by rnburn** [comment]
Looks good. I only had a few small suggestions.

**Comment by KevinMGranger** [comment]
Changes implemented.

**Pull request merged**